src/coreclr/tools/Common/TypeSystem/Ecma/EcmaMethod.cs

```diff
@@ -36,20 +38,16 @@ private static class MethodFlags
         private MethodDefinitionHandle _handle;
 
         // Cached values
+        private nint _namePointer;
+        private int _nameLength;
```

Comment:
I'm not sure I made updating these thread-safe.

Reply:
You may want to mark _namePointer reads as volatile. Either use Volatile.Read or just mark _namePointer as volatile. The latter may allow you to change the `byte*` that's technically more correct.

Matching read and write should be volatile to avoid reordering according to https://github.com/dotnet/runtime/blob/main/docs/design/specs/Memory-model.md#order-of-memory-operations.